src/main/java/org/hiero/sketch/spreadsheet/BucketSetDescription.java

```diff
@@ -0,0 +1,71 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Contains the MetaData for the buckets held by an histogram
+ */
+public class BucketSetDescription {
+    public final int numOfBuckets;
+    public final boolean equalSizeBukets;
+    public final double minValue;
+    public final double maxValue;
+    public final double[] boundaries; //todo: Change the array to a unmodified list of some kind
```

Comment:
You should have an abstract base class, and two derived classes: one for equal-sized buckets, and one for arbitrary buckets.
At least the derived classes should be called something like OneDBucketSet.